src/ppx/imgui_impl.cpp

```diff
@@ -374,6 +374,39 @@ void ImGuiImplVk::Render(grfx::CommandBuffer* pCommandBuffer)
     ImGui::Render();
     ImGui_ImplVulkan_RenderDrawData(ImGui::GetDrawData(), grfx::vk::ToApi(pCommandBuffer)->GetVkCommandBuffer());
 }
+
+void ImGuiImplVk::ProcessEvent()
+{
+#if defined(PPX_BUILD_XR)
+    ImGuiIO&     io   = ImGui::GetIO();
+    Application* pApp = Application::Get();
+    if (pApp->GetSettings()->xr.enable) {
+        XrComponent& xrComponent = pApp->GetXrComponent();
+
+        bool isMouseDown = xrComponent.GetUIClickState().value_or(false);
+        if (isMouseDown != mSimulatedMouseDown) {
+            mSimulatedMouseDown = isMouseDown;
+            io.AddMouseButtonEvent(0, isMouseDown);
+        }
+
+        std::optional<XrVector2f> cursor = xrComponent.GetUICursor();
+        if (cursor.has_value()) {
+            ImGui::SetMouseCursor(ImGuiMouseCursor_Arrow);
+            io.MousePos = ImVec2{
```

Comment:
could you add some comments on why there is a (x+1)/2? (i guess it is to bring the range from [-1,1] to [0,1]?)

Reply:
Ah, I were wondering why the movement feels a bit odd, it's because the mapping is wrong :(
It should be a mapping from `[-0.5m, +0.5m]` to `[0, swapchain.width]`,

Added a comment